gfop/foodflows.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+import pandas as pd
+from typing import List, Optional
+import os
+from importlib import resources
+import pkg_resources
+
+
+class FoodFlows:
+    def __init__(
+        self,
+        gnps_network: str,
+        sample_types: str,
+        groups_included: List[str],
+        max_hierarchy_level: int = 6,
+    ) -> None:
+        """
+        Initializes the FoodFlows object and creates the flows and processes dataframes needed to visualize the flow of foods in the dataset.
+
+        Args:
+            gnps_network (str): Path to the TSV file generated from classical molecular networking.
+            sample_types (str): One of 'simple', 'complex', or 'all'.
+            groups_included (List): Groups of interest in the molecular network to generate food flows
+            max_hierarchy_level (int): Maximum level for food flows calculation.
+        """
+        # Load GNPS network data
+        self.gnps_network = pd.read_csv(gnps_network, sep="\t")
+        self.sample_types = self._load_sample_types(sample_types)
+        self.groups_included = groups_included
+        self.max_hierarchy_level = max_hierarchy_level
+        self.food_metadata = self._load_food_metadata()
+        # Validate group names
+        self._validate_groups()
+        # Generate flows and processes dataframes
+        self.flows, self.processes = self.generate_foodflows()
+
+    def _validate_groups(self) -> None:
```

Comment:
**question:** The next few methods seem identical as in `foodcounts.py`. Can this code duplication be avoided by reusing functionality from there?